cmd/harmony/main.go

```diff
@@ -159,23 +161,43 @@ func raiseFdLimits() error {
 
 func getHarmonyConfig(cmd *cobra.Command) (harmonyConfig, error) {
 	var (
-		config harmonyConfig
-		err    error
+		config       harmonyConfig
+		err          error
+		migratedFrom string
+		configFile   string
 	)
 	if cli.IsFlagChanged(cmd, configFlag) {
-		configFile := cli.GetStringFlagValue(cmd, configFlag)
-		config, err = loadHarmonyConfig(configFile)
+		configFile = cli.GetStringFlagValue(cmd, configFlag)
+		config, migratedFrom, err = loadHarmonyConfig(configFile)
 	} else {
 		nt := getNetworkType(cmd)
 		config = getDefaultHmyConfigCopy(nt)
 	}
 	if err != nil {
 		return harmonyConfig{}, err
 	}
-	if config.Version != defaultConfig.Version {
+	if migratedFrom != defaultConfig.Version {
 		fmt.Printf("Loaded config version %s which is not latest (%s).\n",
-			config.Version, defaultConfig.Version)
-		fmt.Println("Update saved config with `./harmony dumpconfig [config_file]`")
+			migratedFrom, defaultConfig.Version)
+		stat, _ := os.Stdin.Stat()
+		// Ask to update if only using terminal
+		if stat.Mode()&os.ModeCharDevice != 0 {
+			var readStr string
+			fmt.Println("Do you want to update config to latest version: (Y/n)")
+			fmt.Scanf("%s", &readStr)
+			readStr = strings.TrimSpace(readStr)
+			if len(readStr) > 1 {
+				readStr = readStr[0:1]
+			}
+			readStr = strings.ToLower(readStr)
+			if readStr == "y" || readStr == "" {
+				writeHarmonyConfigToFile(config, configFile)
```

Comment:
Would it be better to leave a copy of the older config file being replaced?

Reply:
Yes, same like config update, will move it to <config_path>.backup